numpy/f2py/tests/test_crackfortran.py

```diff
@@ -139,3 +143,20 @@ class TestCrackFortran(util.F2PyTest):
     def test_gh2848(self):
         r = self.module.gh2848(1, 2)
         assert r == (1, 2)
+
+
+class TestMarkinnerspaces():
+    # issue #14118: markinnerspaces does not handle multiple quotations
+
+    def test_do_not_touch_normal_spaces(self):
+        test_list = ["a ", " a", "a b c", "'abcdefghij'"]
+        for i in test_list:
+            assert_equal(markinnerspaces(i), i)
+
+    def test_one_relevant_space(self):
+        assert_equal(markinnerspaces("a 'b c' \\\' \\\'"), "a 'b@_@c' \\' \\'")
+        assert_equal(markinnerspaces(r'a "b c" \" \"'), r'a "b@_@c" \" \"')
```

Comment:
Add the following tests that check that inner quotes are ignored:
```
assert_equal(markinnerspaces('a \'b c" " d\' e', "a 'b@_@c\"@_@\"@_@d' e"))
assert_equal(markinnerspaces('a "b c\' \' d" e', "a \"b@_@c'@)_@'@_@d\" e"))
```

Btw, to make these tests less error-prone, one can use:
```
test_list = ["a 'b@_@c' \\' \\'", r'a "b@_@c" \" \"',
             "a 'b@_@c\"@_@\"@_@d' e", "a \"b@_@c'@)_@'@_@d\" e"]
for i in test_list:
    assert_equal(markinnerspaces(i.replace('@_@', ' ')), i)
```

Reply:
Added